provider/aci.go

```diff
@@ -1635,6 +1635,37 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 func (p *ACIProvider) getVolumes(pod *v1.Pod) ([]aci.Volume, error) {
 	volumes := make([]aci.Volume, 0, len(pod.Spec.Volumes))
 	for _, v := range pod.Spec.Volumes {
+		// Handle the case for Azure File CSI driver
+		if v.CSI != nil {
+			pv, err := p.resourceManager.GetPersistentVolume(v.Name)
+			if err != nil {
+				return volumes, err
+			}
+			csiSource := pv.Spec.CSI
+
+			if csiSource.NodeStageSecretRef != nil && csiSource.NodeStageSecretRef.Name != "" {
```

Comment:
what if csiSource.NodeStageSecretRef == nil? Is there any validator that will reject this beforehand?

Reply:
I wanted to ask if we should check on the `NodePublishSecretRef` property as well? how to know that? any documentation?

Reply:
After digging more into CSI for pods, only the `NodePublishSecretRef` needed to be checked. In case it's empty (as it's optional), we will not be able to add this volume to the ACI volume list .

ref: https://docs.microsoft.com/en-us/rest/api/container-instances/container-groups/create-or-update#volume